projects/addon-mobile/directives/dropdown-mobile/dropdown-mobile.component.ts

```diff
@@ -0,0 +1,106 @@
+import {DOCUMENT} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    Inject,
+    OnDestroy,
+    ViewEncapsulation,
+} from '@angular/core';
+import {TuiKeyboardService} from '@taiga-ui/addon-mobile/services';
+import {tuiGetNativeFocused, tuiPx} from '@taiga-ui/cdk';
+import {
+    TUI_ANIMATIONS_DURATION,
+    TuiDropdownDirective,
+    tuiFadeIn,
+    tuiSlideInTop,
+} from '@taiga-ui/core';
+
+const GAP = 16;
+
+@Component({
+    selector: 'tui-dropdown-mobile',
+    templateUrl: './dropdown-mobile.template.html',
+    styleUrls: ['./dropdown-mobile.style.less'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    animations: [tuiSlideInTop, tuiFadeIn],
+    host: {
+        '[@tuiFadeIn]': 'animation',
+        '[@tuiSlideInTop]': 'animation',
+        '(mousedown.silent.prevent)': '0',
+        '(document:click.silent.capture)': 'onClick($event)',
+        '(window>scroll.silent.capture)': 'refresh($event.currentTarget.visualViewport)',
+        '(visualViewport>resize.silent)': 'refresh($event.target)',
+        '(visualViewport>scroll.silent)': 'refresh($event.target)',
+    },
+})
+export class TuiDropdownMobileComponent implements OnDestroy {
+    private readonly scrollTop = this.doc.documentElement.scrollTop;
+    private readonly observer = new ResizeObserver(() =>
```

Comment:
SSR ?

Why not `@ng-web-apis` ?

```
import {ResizeObserverService} from '@ng-web-apis/resize-observer';
```

Reply:
We do not need RxJS here. I think SSR is ok as it is used only on mobile when dropdown is opened, which is not going to happen under SSR.